bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java

```diff
@@ -0,0 +1,273 @@
+import com.google.bigtable.repackaged.com.google.cloud.bigtable.grpc.BigtableInstanceName;
+import com.google.bigtable.repackaged.com.google.cloud.bigtable.util.ByteStringer;
+import com.google.bigtable.repackaged.com.google.protobuf.ByteString;
+import com.google.cloud.bigtable.hbase.adapters.Adapters;
+import com.google.cloud.bigtable.hbase.adapters.read.DefaultReadHooks;
+import com.google.cloud.bigtable.hbase.adapters.read.ReadHooks;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * This class defines configuration that a Cloud Bigtable client needs to connect to a user's Cloud
+ * Bigtable cluster; a table to connect to in the cluster; and a filter on the table in the form of
+ * a {@link Scan}.
+ */
+public class CloudBigtableScanConfiguration extends CloudBigtableTableConfiguration {
+
+  private static final long serialVersionUID = 2435897354284600685L;
+
+  /**
+   * Converts a {@link CloudBigtableOptions} object to a {@link CloudBigtableScanConfiguration}
+   * that will perform the specified {@link Scan} on the table.
+   * @param config The {@link CloudBigtableTableConfiguration} object.
+   * @param scan The {@link Scan} to add to the configuration.
+   * @return The new {@link CloudBigtableScanConfiguration}.
+   */
+  public static CloudBigtableScanConfiguration fromConfig(CloudBigtableTableConfiguration config,
+      Scan scan) {
+    CloudBigtableScanConfiguration.Builder builder = new CloudBigtableScanConfiguration.Builder();
+    config.copyConfig(builder);
+    return builder.withScan(scan).build();
+  }
+
+  /**
+   * Builds a {@link CloudBigtableScanConfiguration}.
+   */
+  public static class Builder extends CloudBigtableTableConfiguration.Builder {
+    private Scan scan;
+    private ReadRowsRequest request;
+
+    public Builder() {
+    }
+
+    /**
+     * Specifies the {@link Scan} that will be used to filter the table.
+     * @param scan The {@link Scan} to add to the configuration.
+     * @return The {@link CloudBigtableScanConfiguration.Builder} for chaining convenience.
+     */
+    public Builder withScan(Scan scan) {
+      this.scan = scan;
+      this.request = null;
+      return this;
+    }
+
+    /**
+     * Specifies the {@link ReadRowsRequest} that will be used to filter the table.
+     * @param request The {@link ReadRowsRequest} to add to the configuration.
+     * @return The {@link CloudBigtableScanConfiguration.Builder} for chaining convenience.
+     */
+    public Builder withRequest(ReadRowsRequest request) {
+      this.request = request;
+      this.scan = null;
+      return this;
+    }
+
+    /**
+     * Internal API that allows a Source to configure the request with a new start/stop row range.
+     *
+     * @param startKey The first key, inclusive.
+     * @param stopKey The last key, exclusive.
+     *
+     * @return The {@link CloudBigtableScanConfiguration.Builder} for chaining convenience.
+     */
+    Builder withKeys(byte[] startKey, byte[] stopKey) {
```

Comment:
I think withRange or withKeyRange would be a better name. Also would be good to update similar methods like createSourceWithKeys to use a similar terminology